pkt_trail_fe/apps/users/views.py

```diff
@@ -1,3 +1,53 @@
 from django.shortcuts import render
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+import requests
+import os
 
-# Create your views here.
+
+@api_view(('POST',))
+def authenticate(req):
+    code = req.POST.get('code', None)
+
+    if code is None:
+        return Response(dict(message="Code not provided"),status=status.HTTP_401_UNAUTHORIZED)
+
+    access_token = get_access_token(code)
+
+    if access_token:
+        user_info = get_user_info(access_token)
+        if user_info:
+            return Response(dict(user=user_info), status=status.HTTP_200_OK)
+
+        return Response(dict(message="User not found"), status=status.HTTP_404_NOT_FOUND)
+
+    return Response(dict(message="Invalid code"),status=status.HTTP_400_BAD_REQUEST)
+
+
+def get_access_token(code):
```

Comment:
if `GITHUB_CLIENT_ID` or `GITHUB_CLIENT_SECRET` is `None`, you should return `None`. Don't make an API call.